File: onlinejudge/programmers/1/K번째 수.go
import (
	"sort"
)

func solution(array []int, commands [][]int) []int {
	answer := make([]int, len(commands))
	index := 0

	for _, element := range commands {
		startIndex := element[0] - 1
		endIndex := element[1]
		targetIndex := element[2] - 1

		result := make([]int, endIndex-startIndex)
		copy(result, array[startIndex:endIndex])

		sort.Ints(result)
        
		answer[index] = result[targetIndex]
		index++
	}

	return answer
}

File: onlinejudge/programmers/1/K번째 수.java
import java.util.*;

class Solution {
    public int[] solution(int[] array, int[][] commands) {
        int[] result = new int[commands.length];
        int index = 0;
        
        for (int[] element : commands) {
            int startIndex = element[0] - 1;
            int endIndex = element[1];
            int targetIndex = element[2] - 1;
            
            int[] answer = new int[endIndex - startIndex];
            int answerIndex = 0;
            for (int i = startIndex; i < endIndex; i += 1) {
                answer[answerIndex] = array[i];
                answerIndex++;
            }
            
            Arrays.sort(answer);
            result[index] = answer[targetIndex];
            index++;
        }
        
        return result;
    }
}
